src/evo/deterministicmns.cpp

```diff
@@ -928,7 +928,8 @@ bool CDeterministicMNManager::BuildNewListFromBlock(const CBlock& block, gsl::no
                 newState->ResetOperatorFields();
                 newState->BanIfNotBanned(nHeight);
                 // we update pubKeyOperator here, make sure state version matches
-                newState->nVersion = opt_proTx->nVersion;
+                // Make sure we don't accidentally downgrade the state version if using version after basic BLS
+                newState->nVersion = opt_proTx->nVersion > ProTxVersion::BasicBLS ? newState->nVersion : opt_proTx->nVersion;
```

Comment:
shouldn't this be
```suggestion
                newState->nVersion = newState->nVersion > ProTxVersion::BasicBLS ? newState->nVersion : opt_proTx->nVersion;
```
to match the comment above?

Reply:
You're right. Resolved in latest push!